system/lib/gl/webgl2_compute.c

```
@@ -0,0 +1,157 @@
+ASYNC_GL_FUNCTION_1(EM_FUNC_SIG_VI, void, glBindProgramPipeline, GLuint);
+RET_SYNC_GL_FUNCTION_1(EM_FUNC_SIG_VI, GLboolean, glIsProgramPipeline, GLuint);
+ASYNC_GL_FUNCTION_2(EM_FUNC_SIG_VII, void, glActiveShaderProgram, GLuint, GLuint);
+ASYNC_GL_FUNCTION_3(EM_FUNC_SIG_VIII, void, glUseProgramStages, GLuint, GLbitfield, GLuint);
+ASYNC_GL_FUNCTION_1(EM_FUNC_SIG_VI, void, glValidateProgramPipeline, GLuint);
+VOID_SYNC_GL_FUNCTION_4(EM_FUNC_SIG_VIIII, void, glGetProgramPipelineInfoLog, GLuint, GLsizei, GLsizei *, GLchar *);
+VOID_SYNC_GL_FUNCTION_3(EM_FUNC_SIG_VIII, void, glGetProgramPipelineiv, GLuint, GLenum, GLint *);
+ASYNC_GL_FUNCTION_3(EM_FUNC_SIG_VIII, void, glProgramUniform1i, GLuint, GLint, GLint);
+ASYNC_GL_FUNCTION_4(EM_FUNC_SIG_VIIII, void, glProgramUniform2i, GLuint, GLint, GLint, GLint);
+ASYNC_GL_FUNCTION_5(EM_FUNC_SIG_VIIIII, void, glProgramUniform3i, GLuint, GLint, GLint, GLint, GLint);
+ASYNC_GL_FUNCTION_6(EM_FUNC_SIG_VIIIIII, void, glProgramUniform4i, GLuint, GLint, GLint, GLint, GLint, GLint);
+ASYNC_GL_FUNCTION_3(EM_FUNC_SIG_VIII, void, glProgramUniform1ui, GLuint, GLint, GLuint);
+ASYNC_GL_FUNCTION_4(EM_FUNC_SIG_VIIII, void, glProgramUniform2ui, GLuint, GLint, GLuint, GLuint);
+ASYNC_GL_FUNCTION_5(EM_FUNC_SIG_VIIIII, void, glProgramUniform3ui, GLuint, GLint, GLuint, GLuint, GLuint);
+ASYNC_GL_FUNCTION_6(EM_FUNC_SIG_VIIIIII, void, glProgramUniform4ui, GLuint, GLint, GLuint, GLuint, GLuint, GLuint);
+ASYNC_GL_FUNCTION_3(EM_FUNC_SIG_VIIF, void, glProgramUniform1f, GLuint, GLint, GLfloat);
+ASYNC_GL_FUNCTION_4(EM_FUNC_SIG_VIIFF, void, glProgramUniform2f, GLuint, GLint, GLfloat, GLfloat);
+ASYNC_GL_FUNCTION_5(EM_FUNC_SIG_VIIFFF, void, glProgramUniform3f, GLuint, GLint, GLfloat, GLfloat, GLfloat);
+ASYNC_GL_FUNCTION_6(EM_FUNC_SIG_VIIFFFF, void, glProgramUniform4f, GLuint, GLint, GLfloat, GLfloat, GLfloat, GLfloat);
+
+void glProgramUniform1iv(GLuint program, GLint location, GLsizei count, const GLint *value)
+{
+  // TODO
```

perhaps these should `abort()` to avoid silent failure confusion?